src/lib/common/globals.cpp

```diff
@@ -426,24 +427,162 @@ int64_t parse8601(const std::string& s)
   return accumulated;
 }
 
+
+
+/*****************************************************************************
+*
+* timezoneOffset -
+*
+* Returns the time offset corresponding to a given timezone. Only the ISO8601 timezonoe
+* formats are supported (https://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators):
+*
+* <time>Z
+* <time>±hh:mm
+* <time>±hhmm
+* <time>±hh
+*
+* The value -1 is used as "wrong timezone" (note that no timezone corresponds to an
+* offset of just one negative second)
+*
+*/
+static int timezoneOffset(const char* tz)
+{
+  // Trying the <time>Z format
+  if (strcmp(tz, "Z") == 0)
+  {
+    return 0;
+  }
+
+  // All other cases start by + or -
+  if ((tz[0] != '+') && (tz[0] != '-'))
+  {
+    return -1;
+  }
+
+  bool positive = (tz[0] == '+');
+  int offset = -1;
+  int h, m;
+
+  if (sscanf(tz + 1, "%2d:%2d", &h, &m) == 2)      // Trying the <time>±hh:mm format
+  {
+    offset = h * 60 * 60 + m * 60;
+  }
+  else if (sscanf(tz + 1, "%2d%2d", &h, &m) == 2)  // Trying the <time>±hhmm format
+  {
+    offset = h * 60 * 60 + m * 60;
+  }
+  else if (sscanf(tz + 1, "%2d", &h) == 1)         // Trying the <time>±hh format
+  {
+    offset = h * 60 * 60;
+  }
+
+  if (offset == -1)
+  {
+    // wrong timezone
+    return -1;
+  }
+
+  if (positive)
+  {
+    return offset;
+  }
+  else
+  {
+    return -offset;
+  }
+}
+
+
+
 /*****************************************************************************
 *
 * parse8601Time -
 *
-* This is common code for Duration and Throttling (at least)
+* This is common code for Duration and Throttling (at least).
+*
+* Based in http://stackoverflow.com/questions/26895428/how-do-i-parse-an-iso-8601-date-with-optional-milliseconds-to-a-struct-tm-in-c
 *
 */
-int64_t parse8601Time(const std::string& s)
+int64_t parse8601Time(const std::string& ss)
 {
-  struct tm   tm = {0};
-  const char* p;
+  int y, M, d, h, m;
+  char tz[10];
+  float s;
+
```

Comment:
Our style guide says: one variable per line.
And, horizontal indents not to be forgotten.

Also (just a matter of taste), why not initialize h, m and s to zero to avoid doing that in each `if`  ?
Lots of lines to be saved there. The entire if/else if/ ,,, could become a if (() || () ||() ...